src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Text.RegularExpressions
+{
+    [Serializable]
+    internal sealed class RegexParseException : ArgumentException
+    {
+        private readonly RegexParseError _error;
+
+        /// <summary>
+        /// The error that happened during parsing.
+        /// </summary>
+        public RegexParseError Error => _error;
+
+        /// <summary>
+        /// The offset in the supplied pattern.
+        /// </summary>
+        public int Position { get; }
```

Comment:
Since users currently can't get to this (without reflection) maybe include it in the message?